### PR TITLE
Test: Do not remove xpack templates when cleaning

### PR DESCRIPTION
**Pull request by nik9000**
At the end of every `ESRestTestCase` we clean the cluster which includes
deleting all of the templates. If xpack is installed it'll automatically
recreate a few templates every time they are removed. Which is slow.

This change stops the cleanup from removing the xpack templates. It cuts
the time to run the docs tests more than in half and it probably saves a
bit more time on other tests as well.

### CODE REVIEW

**Comment by nik9000** [comment]
@hub-cap, this is another one in the line after the `hasXPack` change.

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -577,4 +594,31 @@ protected static boolean aliasExists(String index, String alias) throws IOExcept
         assertNotNull(responseEntity);
         return responseEntity;
     }
+
+    /**
+     * Is this template one that is automatically created by xpack? Deleting
+     * these templates doesn't hurt anything but it makes the tests run more
+     * slowly because it then has to wait for the templates to be recreated.
```
Comment (hub-cap) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: this comment seems kind of odd for a true/false isXXX method :) It cares not about a delete.

**Review by hub-cap — APPROVED** [review]
LGTM minor comment nit

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 2]
```diff
@@ -281,11 +281,18 @@ private void wipeCluster() throws IOException {
                 Request request = new Request("GET", "_cat/templates");
                 request.addParameter("h", "name");
                 String templates = EntityUtils.toString(adminClient().performRequest(request).getEntity());
-                for (String template : templates.split("\n")) {
-                    if (isXPackTemplate(template)) continue;
-                    adminClient().performRequest(new Request("DELETE", "_template/" + template));
+                if (false == "".equals(templates)) {
+                    for (String template : templates.split("\n")) {
+                        if (isXPackTemplate(template)) continue;
+                        if ("".equals(templates)) {
```
Comment (hub-cap) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: should this be teh singular `template` and not the `list,of,templates`, `templates` ?

**Comment by nik9000** [comment]
Thanks for all the reviews @hub-cap! Merged to master and backporting now.

**Comment by nik9000** [comment]
Backported to 6.x after a passing build!

**Pull request merged**